plugins/modules/ndo_fabric_span_session_source.py

```diff
@@ -0,0 +1,752 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, Sabari Jaganathan (@sajagana) <sajagana@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_fabric_span_session_source
+short_description: Manage Fabric SPAN Sessions Source on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage Switched Port Analyzer (SPAN) Sessions Source on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.4) and later.
+author:
+- Sabari Jaganathan (@sajagana)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a Fabric Monitoring Access Policy template.
+    - This parameter or O(template_id) is required.
+    type: str
+  template_id:
+    description:
+    - The ID of the template.
+    - The template must be a Fabric Monitoring Access Policy template.
+    - This parameter or O(template) is required.
+    type: str
+  span_session_name:
+    description:
+    - The name of the SPAN Session.
+    - This parameter or O(span_session_uuid) is required.
+    type: str
+  span_session_uuid:
+    description:
+    - The UUID of the SPAN Session.
+    - This parameter or O(span_session_name) is required.
+    type: str
+  name:
+    description:
+    - The name of the SPAN Session source.
+    type: str
+  direction:
+    description:
+    - The direction of the SPAN Session source.
+    - Defaults to O(direction=incoming) when unset during creation.
+    type: str
+    choices: [ incoming, outgoing, both ]
+  span_drop_packets:
+    description:
+    - The SPAN Drop Packets of the SPAN Session source.
+    - Defaults to O(span_drop_packets=false) when unset during creation.
+    - The O(filter_epg) and O(filter_l3out) is not configurable when this parameter O(span_drop_packets=true) set to true.
+    type: bool
+  filter_epg:
+    description:
+    - The Filter EPG of the SPAN Session source.
+    - When the Filter EPG is specified in the configuration, the Filter L3Out will be removed.
+    - Filter L3Out and Filter EPG cannot be configured simultaneously.
+    - Providing an empty dictionary O(filter_epg={}) will remove the filter l3out from the SPAN Session source.
+    type: dict
+    suboptions:
+      uuid:
+        description:
+        - The UUID of the EPG used to configure the Filter EPG.
+        - This parameter or O(filter_epg.epg) is required.
+        type: str
+        aliases: [ epg_uuid ]
+      epg:
```

Comment:
Do we want to name this `epg` or something like `reference`? I personally find the filter_epg.epg a bit redundant. In my PR I have a `port_channel` dict, which would then be `port_channel.port_channel.some_key`.